### PR TITLE
fix(secu): sanitize host dashboard

### PR DESCRIPTION
**Pull request by adr-mo**
Backport of https://github.com/centreon/centreon/pull/9372

**Fixes** # (issue)

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 19.10.x
- [ ] 20.04.x
- [ ] 20.10.x
- [ ] 21.04.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

## Checklist

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: www/include/reporting/dashboard/common-Func.php [hunk 1]
```diff
@@ -38,29 +38,58 @@
...
+    } elseif (isset($_GET['EndDate'])) {
+        $end_date = filter_var($_GET['EndDate'], FILTER_SANITIZE_STRING);
+    }
+
     $interval = array(0, 0);
-    if ($period_choice == "custom" && $start_date != "" && $end_date != "") {
-        $period = "";
+
+    if (
+        $period_choice == 'custom' &&
```
Comment (sc979) [inline www/include/reporting/dashboard/common-Func.php:71]:
```suggestion
    if ($period_choice == 'custom' &&
```

File: www/include/reporting/dashboard/common-Func.php [hunk 2]
```diff
@@ -38,29 +38,58 @@
...
         $interval = getDateSelectCustomized($start_date, $end_date);
     } else {
         $interval = getDateSelectPredefined($period);
     }
-    $start_date = $interval[0];
-    $end_date = $interval[1];
+
+    list($start_date, $end_date) = $interval;
+
     return(array($start_date, $end_date));
```
Comment (sc979) [inline www/include/reporting/dashboard/common-Func.php:93]: why not return $interval ?
if interval contains only $start_date and $end_date

File: www/include/reporting/dashboard/common-Func.php [hunk 3]
```diff
@@ -38,29 +38,58 @@
...
-    $end_date = 0;
-    $start_date = 0;
-    $start_date = (isset($_POST["StartDate"])) ? $_POST["StartDate"] : "";
-    $start_date = (isset($_GET["start"])) ? $_GET["start"] : $start_date;
-    $end_date = (isset($_POST["EndDate"])) ? $_POST["EndDate"] : "";
-    $end_date = (isset($_GET["end"])) ? $_GET["end"] : $end_date;
+    $period = '';
+    $start_date = '';
+    $end_date = '';
+    $period_choice = '';
```
Comment (sc979) [inline www/include/reporting/dashboard/common-Func.php:44]: could you convert these variables to camelCase

**Review by sc979 — CHANGES_REQUESTED** [review]

**Comment by adr-mo** [comment]
@sc979 Sure :+1:

**Comment by adr-mo** [comment]
:heavy_check_mark: Done :)

File: www/include/reporting/dashboard/common-Func.php (hunk 2 above)
Comment (adr-mo) [reply www/include/reporting/dashboard/common-Func.php]: Did not really thought it through :+1: 
I'll keep it this way for now

**Review by sc979 — APPROVED** [review]

**Review by kduret — APPROVED** [review]

**Pull request merged**